src/Blueprints/ShippingZoneBlueprint.php

```diff
@@ -41,6 +43,44 @@ public function __invoke()
                                 'validate' => 'required|exists:butik_shipping_profiles,slug',
                             ],
                         ],
+                        [
+                            'handle' => 'countries',
+                            'field' => [
+                                'type' => 'select',
+                                'options' => Countries::getNames(app()->getLocale()),
+                                'clearable' => false,
+                                'multiple' => true,
+                                'searchable' => true,
+                                'taggable' => false,
+                                'push_tags' => false,
+                                'cast_booleans' => false,
+                                'localizable' => false,
+                                'listable' => true,
+                                'display' => 'Countries',
+                                'validate' => ['required', 'array',
+                                    function ($attribute, $value, $fail) {
+                                        foreach ($value as $country_code) {
+                                            if (! Countries::exists($country_code)) {
+                                                $fail('Invalid country code: ' . $country_code);
+                                            }
+                                        }
+                                    },
+                                    function ($attribute, $value, $fail) {
+                                        if(ShippingZone::all()
+                                            ->filter(function($shipping_zone) use ($value) {
+                                                foreach ($value as $country_code) {
+                                                    if ($shipping_zone->countries->contains($country_code)) {
+                                                        return true;
+                                                    }
+                                                }
+                                            })
+                                            ->count() > 1) {
+                                            $fail('One of the countries is already added to another shipping zone.');
```

Comment:
We should abstract this validation into it's own validation rule

Reply:
Sure, we could do it. However, I don't see a case for re-use here. There are 3 country-related validators used:

1. For single country code: used to validate customer's shipping address.
2. For array of country codes: to validate input on countries select field in `ShippingZone`.
3. For country codes used in other `ShippingZone` records: to avoid adding same country to multiple ShippingZones.

None of these are used twice right now.